src/array/cpu/concurrent_id_hash_map.cc

```diff
@@ -111,7 +111,9 @@ IdArray ConcurrentIdHashMap<IdType>::Init(
   parallel_for(num_seeds, num_ids, kGrainSize, [&](int64_t s, int64_t e) {
     size_t count = 0;
     for (int64_t i = s; i < e; i++) {
-      Insert(ids_data[i], &valid, i);
+      if (Insert(ids_data[i])) {
```

Comment:
I am assuming each i will only be accessed once:
It can be simplified to:
valid[i] = Insert(ids_data[i]);
count += valid[i];

This is actually better since the existing code in L107 assumes the valid will be initiated to 0, which might not be true for all c++ compiler.

Reply:
It makes sense. changed to this style.